From bfaff5ded919c6d872c5954ee4664c83863ccb2f Mon Sep 17 00:00:00 2001
From: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
Date: Thu, 16 Feb 2017 21:40:03 +0100
Subject: [PATCH] Add osc14M clock

Signed-off-by: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
---
 arch/arm/boot/dts/sun5i.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/sun5i.dtsi b/arch/arm/boot/dts/sun5i.dtsi
index 36b69ab..dbc14d4 100644
--- a/arch/arm/boot/dts/sun5i.dtsi
+++ b/arch/arm/boot/dts/sun5i.dtsi
@@ -79,6 +79,12 @@
 			compatible = "fixed-clock";
 			clock-frequency = <0>;
 		};
+		
+		osc14M: osc14M_clk {
+			#clock-cells = <0>;
+			compatible = "fixed-clock";
+			clock-frequency = <14745600>;
+		};
 
 		osc24M: clk@01c20050 {
 			#clock-cells = <0>;
-- 
1.9.1

