From c57cf712caaf9e9826c593e46687506dd5834162 Mon Sep 17 00:00:00 2001
From: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
Date: Tue, 28 Mar 2017 17:38:05 +0200
Subject: [PATCH] Add Osc 10M

Signed-off-by: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
---
 arch/arm/boot/dts/sun5i.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/sun5i.dtsi b/arch/arm/boot/dts/sun5i.dtsi
index 36b69ab..7bccf42 100644
--- a/arch/arm/boot/dts/sun5i.dtsi
+++ b/arch/arm/boot/dts/sun5i.dtsi
@@ -79,6 +79,12 @@
 			compatible = "fixed-clock";
 			clock-frequency = <0>;
 		};
+		
+		osc10M: osc10M_clk {
+			#clock-cells = <0>;
+			compatible = "fixed-clock";
+			clock-frequency = <10000000>;
+		};
 
 		osc24M: clk@01c20050 {
 			#clock-cells = <0>;
-- 
1.9.1

