From 22ea3d0b811adc53734bf64f6d66dc6ca7df263a Mon Sep 17 00:00:00 2001
From: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
Date: Wed, 26 Apr 2017 21:16:44 +0200
Subject: [PATCH] ad525x_dpot i2c Add device tree support

Signed-off-by: Pierre-Jean TEXIER <pjtexier@amplitude-systemes.com>
---
 drivers/misc/ad525x_dpot-i2c.c | 35 ++++++++++++++++++++++++++++++++++-
 1 file changed, 34 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/ad525x_dpot-i2c.c b/drivers/misc/ad525x_dpot-i2c.c
index 4f83200..ef26673 100644
--- a/drivers/misc/ad525x_dpot-i2c.c
+++ b/drivers/misc/ad525x_dpot-i2c.c
@@ -8,7 +8,7 @@
 
 #include <linux/i2c.h>
 #include <linux/module.h>
-
+#include <linux/of.h>
 #include "ad525x_dpot.h"
 
 /* I2C bus functions */
@@ -103,9 +103,42 @@ static const struct i2c_device_id ad_dpot_id[] = {
 };
 MODULE_DEVICE_TABLE(i2c, ad_dpot_id);
 
+#ifdef CONFIG_OF
+static const struct of_device_id ad_dpot_of_match[] = {
+	{ .compatible = "adi,ad5258", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5259", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5251", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5252", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5253", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5254", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5255", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5241", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5242", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5243", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5245", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5246", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5247", .data = (void *) AD5160_ID, },	
+	{ .compatible = "adi,ad5248", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5280", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5282", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,adn2860", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5273", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5161", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5171", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5170", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5172", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5173", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5272", .data = (void *) AD5160_ID, },
+	{ .compatible = "adi,ad5274", .data = (void *) AD5160_ID, },
+	{/* Sentinel */}
+};
+MODULE_DEVICE_TABLE(of, ad_dpot_of_match);
+#endif
+
 static struct i2c_driver ad_dpot_i2c_driver = {
 	.driver = {
 		.name	= "ad_dpot",
+		.of_match_table	= of_match_ptr(ad_dpot_of_match),
 	},
 	.probe		= ad_dpot_i2c_probe,
 	.remove		= ad_dpot_i2c_remove,
-- 
1.9.1

